### PR TITLE
ENH: add _repr_html_ method in the classes Info and BaseRaw

### PR DESCRIPTION
**Pull request by vagechirkov**
Reference issue #6846.

Reuse `report.py` code in order to produce html reports within the Info class.

Here is the output of in the IPython notebook:

image

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
so far so good !

**Review by larsoner — COMMENTED** [review]
Can you modify `tutorials/intro/plot_30_info.py` to have a line like:
```
info
```
at the end of a code block? Then CircleCI will render it and we'll get to see the `_repr_html_` in the built doc of this PR

**Comment by vagechirkov** [comment]
Here is the output of in the IPython notebook:

image
image

File: mne/data/html_templates.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+from ..externals.tempita import Template
+
+
+info_template = Template(u"""
```
Comment (drammock) [inline mne/data/html_templates.py:4]:
```suggestion
info_template = Template("""
```

File: mne/data/html_templates.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        <td>{{u'%0.2f' % info['highpass']}} Hz</td>
+    </tr>
+     <tr>
+        <th>Lowpass</th>
+        <td>{{u'%0.2f' % info['lowpass']}} Hz</td>
+    </tr>
+</table>
+""")
+
+raw_template = Template(u"""
```
Comment (drammock) [inline mne/data/html_templates.py:60]:
```suggestion
raw_template = Template("""
```

**Review by drammock — APPROVED** [review]
just two tiny bits of Python2 cruft.  This looks great!

**Comment by drammock** [comment]
Here's the CircleCI rendering: https://26281-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/intro/plot_30_info.html#dropping-channels-from-an-info-object

**Comment by drammock** [comment]
@vagechirkov can you mark this as "ready for review"?  Or are you still considering it as a draft?

**Comment by larsoner** [comment]
Awesome, thanks @vagechirkov !

**Comment by agramfort** [comment]
thx heaps @vagechirkov !

a few suggestions to go further. You see that for raw in the report you only show the info from raw. I think
should show show more. Typically the duration of the recording. Also maybe the files it corresponds to (raw._filenames attribute).

other thoughts anyone?

**Pull request merged**